### ISSUE TITLE
Pull Dual Pricing into the core

### ISSUE THREAD

**Issue by scottcwilson** [zencart/zencart#2678]
This mod is incredibly useful and would represent a significant new capability for Zen Cart.

**Comment by wiztechinc** [comment]
When someone needs it, it's great, no question. I have very few carts that have used it though, definitely less than 10%

**Comment by judygunderson** [comment]
I think Admin Login as Customer would be great if it is not already included!

**Comment by scottcwilson** [comment]
Judy, I quite agree - would you please open a separate issue about that so we can keep this ticket on the Dual Pricing topic?  Thanks.

**Comment by drbyte** [comment]
Keep in mind that pulling mods into the core implicitly expects "us" to do all the support of the feature and maintain the feature. 
Are you volunteering to take on that burden for the foreseeable future?

**Comment by scottcwilson** [comment]
For as long as I can draw breath, sir.

**Comment by wiztechinc** [comment]
> For as long as I can draw breath, sir.

And we hope you will be around a long long time. But I assume you are all aware how many have left or died - in my case, I ran into major medical issues and have simply not had the time or ability to keep up with updating mods that I wrote. (hee, hee, I'm back!)  Chris has a very valid point if one is looking long term.

**Comment by scottcwilson** [comment]
The Good Lord willing and the creek don't rise, I have many releases left in me. :)

**Comment by wiztechinc** [comment]
> > not had the time or ability to keep up with updating mods that I wrote
> 
> Me neither....hence I think the more in the core the better, so "problems" are more in the public domain/in more peoples faces and more likely to get attention/fixed.
> Once you have touched a Plugin/mod it sticks to you and everyone thinks its your baby/the community input disappears.

Boy oh boy are you right! After trying to do the right thing and upload changes to a mod, everyone thinks you got the answers when you just made some minor fixes/changes. One of the chief reasons I pulled back from fixing other mods. I just keep them to myself now.

**Comment by torvista** [comment]
I don't use either yet, but absolutely need to. My sales have gradually shifted to 90% trade, and shops are now **expecting** to be forced to buy online from their distributors.
To summarise from what I have just been reading:
stock Group Discount allows a global discount based on the group the customer belongs to.
Inadequate.
mod Dual Pricing allows four price levels per product and customers can be assigned any of them.
This will become the third essential mod for me (Advanced Shipper and Products Options Stock since you asked), so yes I would vote this in the core.

**Comment by lankeeyankee** [comment]
me too, we've been using it for 10+ years. I would call it essential

**Comment by dakanji** [comment]
Rather than adding the whole kitchen sink to the core, what should happen is that some modules should be officially supported and actually, the core should be as lean as possible. 

A subtle but important difference. Still means needing to support them but this will simplify development as a whole.

**Comment by torvista** [comment]
> the core should be as lean as possible
I feel that is valid from a developer point of view.
But ZC is not for developers, it is for business-owners and in my view should have a longer list of core functionality than the competition, if it wants to compete.
There needs to be more input requested from the forum as to what should be included, most users have no need to venture into Github and wont comment here.

**Comment by dakanji** [comment]
Everything that has been said is still easily achievable using officially supported modules.

Once something is a supported plugin, the work to support it needs to be in the core and this is what the core should be focused on. So shouldn't be a need to keep on patching for new releases.

With respect to concerned about people having to stumble upon such, they could, and should, be distributed with the program by default.

The advantage is that additional functionality, bug/security fixes can be rolled out in a modular fashion without having to wait for the whole thing to be updated or for threads with patches on the forum etc that someone hopefully stumbles upon.

In such cases, a new version of the affected module(s) is rolled out and job done. Particularly if they dial home and tell the user to upgrade.

Far easier for developers/maintainers and beneficial for users.

Anyway, just throwing out a thought for consideration. Point is not just on this one particular functionality but a general paradigm.

**Comment by scottcwilson** [comment]
Possible 1.5.8 or future feature.

**Comment by torvista** [comment]
Hopefully I'll get my false teeth stuck into it before then!

**Comment by torvista** [comment]
Some months ago I looked at this and didn't like it.
I made my own from scratch using observers all the way. I still didn't like it as I found the manipulation of pricing when you get into the discounts to be a world of pain and so ended with a non-ideal/basic solution that is adequate for my needs but not for public consumption/release.
I still think the concept is valid for inclusion, but not until the pricing code is rewritten...is there a perceived need/plan for that?
